### PR TITLE
remove script Download-GitExtensions.ps1

### LINKED ISSUE #28: Use GitExtensions.Extensiblity in plugin project
Replace original GE download scripts with dependency on `GitExtensions.Extensibility`.

### PR DESCRIPTION
**Pull request by mast-eu**
Resolves #28 

## Proposed changes

* Add package "GitExtensions.Extensibility" as dependency and use it's pre- and postbuild events to replace script `tools/Download-GitExtensions.ps1`
* Modify `appveyor.yml` and `tools/Prepare-Release.ps1`, because the *.nupkg artifact went missing on AppVeyor after some recent build worker image update.
See https://ci.appveyor.com/project/gitextensions/gitextensions-pluginmanager/builds/27138736/artifacts vs. https://ci.appveyor.com/project/gitextensions/gitextensions-pluginmanager/builds/27838894/artifacts 
Both are builds of https://github.com/gitextensions/gitextensions.pluginmanager/commit/91c5b32b53fb8dee3a93e343a2794d7543451f50, the first ran on 2019-09-03 and the second ran just now.

## Twin PR
* https://github.com/gitextensions/gitextensions.plugintemplate/pull/10

### CODE REVIEW

**Comment by mast-eu** [comment]
@maraf: Please have a look.

File: src/GitExtensions.PluginManager/GitExtensions.PluginManager.csproj [hunk 1]
```diff
@@ -13,25 +13,33 @@
...
+  <ItemGroup>
+    <PackageReference Include="GitExtensions.Extensibility" Version="0.1.1" />
+  </ItemGroup>
+
+  <PropertyGroup>
+    <GitExtensionsDownloadPath>..\..\..\gitextensions.shared</GitExtensionsDownloadPath> <!-- path is relative to $(ProjectDir) -->
+    <GitExtensionsReferenceVersion>latest</GitExtensionsReferenceVersion> <!-- 'latest' or 'v3.1' (= tag from GitHub releases) or 'v3.1.0.5877' (= build number from AppVeyor)-->
+    <GitExtensionsReferenceSource>GitHub</GitExtensionsReferenceSource> <!-- 'GitHub' or 'AppYevor' -->
+    <GitExtensionsPath></GitExtensionsPath> <!-- for local builds (no download) -->
+  </PropertyGroup>
```
Comment (maraf) [inline src/GitExtensions.PluginManager/GitExtensions.PluginManager.csproj]: What if I don't have the folder `..\..\..\gitextensions.shared` locally? Will it still download GE for the first time?

Shoouldn'd this personal setting be placed in a `.csproj.user`?

Comment (mast-eu) [reply src/GitExtensions.PluginManager/GitExtensions.PluginManager.csproj]: > What if I don't have the folder `..\..\..\gitextensions.shared` locally? Will it still download GE for the first time?

Yes. The folder will be automatically created if it doesn't exist.

> Shoouldn'd this personal setting be placed in a `.csproj.user`?

Good idea. I'll put them into a .user file and add it to .gitignore.  this way we can avoid accidential commits of local configurations.

Comment (mast-eu) [reply src/GitExtensions.PluginManager/GitExtensions.PluginManager.csproj]: Done.

**Comment by mast-eu** [comment]
Just a friendly reminder that this PR is blocked because it has no approving review.
Or is there something else I should modify before merging?

**Comment by mast-eu** [comment]
No problem. Thank you.

**Pull request merged**